velox/functions/prestosql/RandN.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "folly/Random.h"
+#include "velox/functions/Macros.h"
+
+namespace facebook::velox::functions {
+
+template <typename T>
+struct RandNFunction {
+  static constexpr bool is_deterministic = false;
+
+  template <typename TOutput, typename TInput = TOutput>
```

Comment:
It might be simpler to move this code into Rand.h and just add multiple 'call' methods, one for each supported type.